homeassistant/components/solaredge/sensor.py

```diff
@@ -319,6 +355,68 @@ def update(self):
         _LOGGER.debug("Updated SolarEdge inventory: %s, %s", self.data, self.attributes)
 
 
+class SolarEdgeEnergyDetailsService(SolarEdgeDataService):
+    """Get and update the latest power flow data."""
+
+    def __init__(self, api, site_id):
+        """Initialize the power flow data service."""
+        super().__init__(api, site_id)
+
+        self.unit = None
+
+    @Throttle(ENERGY_DETAILS_DELAY)
+    def update(self):
+        """Update the data from the SolarEdge Monitoring API."""
+        try:
+            now = datetime.now()
+            today = date.today()
+            midnight = datetime.combine(today, datetime.min.time())
+            data = self.api.get_energy_details(
+                self.site_id,
+                midnight,
+                now.strftime("%Y-%m-%d %H:%M:%S"),
+                meters=None,
+                time_unit="DAY",
+            )
+            energy_details = data["energyDetails"]
+        except KeyError:
+            _LOGGER.error("Missing power flow data, skipping update")
+            return
+        except (ConnectTimeout, HTTPError):
+            _LOGGER.error("Could not retrieve data, skipping update")
+            return
+
+        if "meters" not in energy_details:
+            _LOGGER.debug(
+                "Missing meters in energy details data. Assuming site does not have any"
+            )
+            return
+
+        self.data = {}
+        self.attributes = {}
```

Comment:
You should not assign `self.attributes` until the end of the function when it has all the info. 

Consider replacing this class with [DataUpdateCoordinator](https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities)

Reply:
Thanks for your feedback. 
I'm sorry but I'm not so skilled to replace all the classes. I only extended what was already in place.

Each classes, for what I've understand,  is recalled on SolarEdgeSensorFactory, where sensor is created with create_sensor. So every data and attributes's variables are returned to it. If you check each classes got same data/attributes assignment, before are filled.  

So to accomplish what you've asked I guess it's need so much work to recode all the plugin, that's beyond my capability.